tests/test_edit.rs

```diff
@@ -450,6 +450,67 @@ fn test_sort_values() {
 "#);
 }
 
+#[test]
+fn test_multiple_positions() {
+    given(r#"
+        [package]
+        [dependencies]
+        [dependencies.opencl]
+        [dev-dependencies]"#
+    ).running(|root| {
+        for (_, table) in root.iter_mut() {
+            as_table!(table).set_position(2)
+        }
+    }).produces_in_original_order(r#"
+        [package]
+        [dependencies]
+        [dev-dependencies]
+        [dependencies.opencl]
+"#);
+}
+
+#[test]
+fn test_multiple_zero_positions() {
+    given(r#"
+        [package]
+        [dependencies]
+        [dependencies.opencl]
+        a=""
+        [dev-dependencies]"#
+    ).running(|root| {
+        for (_, table) in root.iter_mut() {
+            as_table!(table).set_position(0)
+        }
+    }).produces_in_original_order(r#"
+        [package]
+        [dependencies]
+        [dev-dependencies]
+        [dependencies.opencl]
+        a=""
+"#);
+}
+
+#[test]
+fn test_multiple_max_usize_positions() {
+    given(r#"
+        [package]
+        [dependencies]
+        [dependencies.opencl]
+        a=""
+        [dev-dependencies]"#
+    ).running(|root| {
+        for (_, table) in root.iter_mut() {
+            as_table!(table).set_position(usize::MAX)
+        }
+    }).produces_in_original_order(r#"        [dependencies.opencl]
+        a=""
+
+        [package]
+        [dependencies]
+        [dev-dependencies]
+"#);
+}
```

Comment:
The formatting in the mentioned docs is actually expected, because we treat spaces and comment to the left/right/above a table's header as table's decoration. 

The ordering of tables appears wrong because you only set position for direct children of root (all tables except `dependencies.opencl`.

I think it would be better to be explicit about which tables have their position altered.